traitlets/config/loader.py

```diff
@@ -709,71 +834,43 @@ def load_config(self, argv=None, aliases=None, flags=None):
             A list that has the form of sys.argv[1:] which has unicode
             elements of the form u"key=value". If this is None (default),
             then self.argv will be used.
-        aliases : dict
-            A dict of aliases for configurable traits.
-            Keys are the short aliases, Values are the resolved trait.
-            Of the form: `{'alias' : 'Configurable.trait'}`
-        flags : dict
-            A dict of flags, keyed by str name. Values can be Config objects
-            or dicts.  When the flag is triggered, The config is loaded as
-            `self.config.update(cfg)`.
         """
         self.clear()
         if argv is None:
             argv = self.argv
-        if aliases is None:
-            aliases = self.aliases
-        if flags is None:
-            flags = self.flags
+        if aliases:
+            raise ValueError("%s does not support aliases" % self.__class__.__name__)
+        if flags:
+            raise ValueError("%s does not support flags" % self.__class__.__name__)
```

Comment:
I get from this that this is on purpose; so things like `--Foo.Bar a=1 -F b=2` where foo is an alias of `--Foo.Bar` is not handled ? Or does that just mean it's in progress ?

Reply:
This actually doesn't affect command-line parsing in 'real' applications, which use `KVArgParseConfigLoader` below. `KVArgParseConfig` works like this:

1. build an argparse Parser to handle aliases and flags
2. call `parse_known_args` to get alias and flag config,
3. construct one of these `KeyValueConfigLoader`s to handle parsing the remaining *unrecognized* arguments, which should all be arbitrary `--Class.trait=foo` (or unrecognized aliases that should raise).

The result is that these `KeyValueConfigLoader` objects are *only* used to parse full `--Class.trait` args, never aliases or flags. In changing the implementation here to use argparse (mainly to get consistent multiply-specified and `=` vs space handling), I lost the previous support for aliases, which was never used, since these were by definition used only on non-alias/flag options. Rather than complicate things further by reimplementing alias support where it will never be used, I opted to make the lack of support in thus sub-parser explicit.

I'll have a look at docstrings and comments to try to make this clearer.